web/src/main/resources/static/js/tree.js

```diff
@@ -39,15 +35,79 @@ var tree = (function () {
         this.acronym = '';
         this.mainType = '';
         this.color = '';
-        this.nci = '';
-        this.umls = '';
+        this.nci = [];
+        this.umls = [];
+        this.history = ''; // comma delimited string
+    }
+
+    function getOncotreeCodeKeysSortedByName(oncotreeNodeDict) {
+        return Object.keys(oncotreeNodeDict).sort(function(a,b) {
+                var nameA = oncotreeNodeDict[a].name;
+                var nameB = oncotreeNodeDict[b].name;
+                if (nameA < nameB) {
+                    return -1;
+                }
+                if (nameA > nameB) {
+                    return 1;
+                }
+                return 0;
+            });
+    }
+
+    function process_children(parentNode, childData) {
+        // childData is always for a new unique node
+        var childNode = new UniqueTreeNodeDatum();
+        childNode.name = childData.name + " (" + childData.code + ")";
+        childNode.acronym = childData.code;
+
+        if (childData.hasOwnProperty('mainType') && childData.mainType != null && childData.mainType.hasOwnProperty('name') && childData.mainType.name !== '') {
+            childNode.mainType = childData.mainType.name;
+        } else {
+            childNode.mainType = 'Not Available';
```

Comment:
good ... this will now cover the "TISSUE" root node which is not present in tumor_types.txt